homeassistant/components/onvif/config_flow.py

```diff
@@ -51,12 +51,15 @@
 def wsdiscovery() -> list[Service]:
     """Get ONVIF Profile S devices from network."""
     discovery = WSDiscovery(ttl=4)
-    discovery.start()
-    services = discovery.searchServices(
-        scopes=[Scope("onvif://www.onvif.org/Profile/Streaming")]
-    )
-    discovery.stop()
-    return services
+    try:
+        discovery.start()
+        return discovery.searchServices(
+            scopes=[Scope("onvif://www.onvif.org/Profile/Streaming")]
+        )
+    finally:
+        discovery.stop()
+        # Stop the threads started by WSDiscovery since otherwise there is a leak.
+        discovery._stopThreads()  # pylint: disable=protected-access
```

Comment:
Can we refactor the library so we don't need to access protected methods?

Reply:
We don't control the library code and the repo is not active since 2020

Reply:
https://pypi.org/project/WSDiscovery/

Reply:
Also, the problem is already fixed on the repo but never released

Reply:
Annoying. Ok. :+1: